Assets/Scripts/Buddy/BuddyStats.cs

```diff
@@ -204,14 +206,13 @@ public void AdjustHappiness( float deltaHappiness )
 		{
 			_happiness = 0;
 		}
-
-		RecalculateStat();
-
-		if ( _happiness < 0 )
+		else if ( _happiness > 1 )
 		{
-			_happiness = 0;
+			_happiness = 1;
```

Comment:
Should use 1f for this and similar situations. It's a miniscule performance boost and a good habit to get into